com.unity.render-pipelines.core/Runtime/Common/RemoveRange.Extensions.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using JetBrains.Annotations;
+
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// A set of extension methods for collections
+    /// </summary>
+    public static class RemoveRangeExtensions
+    {
+        /// <summary>
+        /// Tries to remove a range of elements from the list.
+        /// </summary>
+        /// <param name="list">The list to remove the range</param>
+        /// <param name="index">The zero-based starting index of the range of elements to remove</param>
+        /// <param name="count">The number of elements to remove.</param>
+        /// <param name="error">The exception raised by the implementation</param>
+        /// <typeparam name="TList">The type of the input list</typeparam>
+        /// <typeparam name="TValue">The value type stored on the list</typeparam>
+        /// <returns>True if succeed, false otherwise</returns>
+        [CollectionAccess(CollectionAccessType.ModifyExistingContent)]
+        [MustUseReturnValue]
+        public static bool TryRemoveRange<TList, TValue>([DisallowNull] this TList list, int index, int count, [NotNullWhen(false)] out Exception error)
+            where TList : IList<TValue>
+        {
+            try
+            {
+                // This branching is inlined at compilation
+                if (typeof(TList) == typeof(List<TValue>))
+                {
+                    // Reinterpret cast, this is safe because we checked the types before
+                    var castedList = __refvalue(__makeref(list), List<TValue>);
```

Comment:
Actually did not even knew about these keywords. IIRC both HDRP and URP has collection package.
Maybe for the consistency we can replace this with Unity.Collections.LowLevel.Unsafe.UnsafeUtility.AddressOf and Unity.Collections.LowLevel.Unsafe.UnsafeUtility.AsRef?

Reply:
The aforementionned API works with custom allocated memory, pointers and only structs in very specialized contexts. 
So they can be used on reference type like ` List`. (And it is safer with this restriction as you can't use it on instances allocated in managed memory accidentaly)